src/util/samplebuffer.h

```diff
@@ -34,64 +34,65 @@ namespace mixxx {
 //
 class SampleBuffer final {
   public:
-    SampleBuffer()
+    constexpr SampleBuffer()
         : m_data(nullptr),
           m_size(0) {
     }
     explicit SampleBuffer(SINT size);
     SampleBuffer(SampleBuffer&) = delete;
     SampleBuffer(SampleBuffer&& that)
-        : m_data(that.m_data),
-          m_size(that.m_size) {
-        that.m_data = nullptr;
-        that.m_size = 0;
+            : m_data(std::exchange(that.m_data, nullptr)),
+              m_size(std::exchange(that.m_size, 0)) {
     }
-    virtual ~SampleBuffer() final;
+    ~SampleBuffer();
 
     SampleBuffer& operator=(SampleBuffer& that) = delete;
-    SampleBuffer& operator=(SampleBuffer&& that) {
-        swap(that);
-        return *this;
-    }
+    SampleBuffer& operator=(SampleBuffer&& that);
```

Comment:
is a particular reason you chose to put the move ctor in the header but not the move-assignment?

Reply:
I move it to the .cpp

Reply:
Well I would've done it the other way around. Seems like the compiler could benefit from knowing the implementation (being able to inline, etc). I'm just curious why you preferred it in the .cpp?

Reply:
I didn't think about it, I just wrote it were my cursor was.

Reply:
At the end of the day, if you build with IPO then it is more or less the same right?

Reply:
Probably, but we currently don't build with LTO afaik, I think we tried but it caused some issues. Especially on a codebase as large as mixxx. We already switched `mold` because linking times are quite long out-of-the-box, using LTO wouldn't help.

Reply:
Then, all the SampleBuffer.cpp could go into the header.

Reply:
If you don't mind, I'll leave it as it is?

Reply:
I doubt that any of the SampleBuffer.cpp is used from a hotspot except fill() and clear() maybe.

Reply:
> Then, all the SampleBuffer.cpp could go into the header.

Frankly, yes...

> If you don't mind, I'll leave it as it is?

I don't really mind tbh, but that kinda feels like its been half done. I think samplebuffer is in a much better state already thanks to your efforts, and this would be just be the last step to make it consistent.

Reply:
I've done it.
Waiting for my computer to build it... taking at least 10 minutes it seems.